src/storage/src/storage_state/async_storage_worker.rs

```diff
@@ -247,81 +249,51 @@ impl<T: Timestamp + Lattice + Codec64 + Display> AsyncStorageWorker<T> {
                                 .open(persist_location.clone())
                                 .await
                                 .expect("error creating persist client");
+                            let diagnostics = Diagnostics {
+                                shard_name: id.to_string(),
+                                handle_purpose: format!("resumption data {}", id),
+                            };
 
                             let mut write_handle = client
                                 .open_writer::<SourceData, (), T, Diff>(
                                     *data_shard,
                                     Arc::new(relation_desc.clone()),
                                     Arc::new(UnitSchema),
-                                    Diagnostics {
-                                        shard_name: id.to_string(),
-                                        handle_purpose: format!("resumption data {}", id),
-                                    },
+                                    diagnostics.clone(),
                                 )
                                 .await
                                 .unwrap();
                             let upper = write_handle.fetch_recent_upper().await;
+
+                            let read_handle = client
```

Comment:
Can't it still happen that a since downgrade on the data shard is rate-limited while a since downgrade on the remap shard goes through? In that case, we'd pick an as of here that would then make the assert in `reclock/compat.rs` fail.

Of course, with the previous code it could also happen that we pick an `as_of` based on the remap shard but then the controller concurrently downgrades that one, right?